tier4/Pilot.Auto
# ROS2 Porting: gnss_poser

# gnss_poser

Awaiting on the following PRs:
* [x] https://github.com/tier4/AutowareArchitectureProposal/pull/69

## Summary

Related PRs:
* https://github.com/tier4/ublox/pull/2
* https://github.com/tier4/AutowareArchitectureProposal/pull/69

Convert the gnss_poser package to ROS2. Requires a number of dependencies which are not in the docker:
* ublox_msgs
* geographic_lib
* geo_pos_conv (already ported)

## Testing

Testing this branch will require some dependencies which are in preinstalled in Foxy:
* GeographicLib (installed via rosdep)
* diagnostic_updater (installed via rosdep)
* ublox_msgs (build and install)

Setup:
1. Checkout PR branch for AutowareArchitectureProposal repository (https://github.com/tier4/AutowareArchitectureProposal/pull/69)
2. Modify the reference to the ublox to point to the PR branch (https://github.com/tier4/ublox/pull/2) - since this has not already been merged
3. Run the vcs tool
```
vcs import src < autoware.proj.repos
```
4. Enter your environment of choice (docker/ADE/etc.)
5. Install rosdep dependencies
```
sudo apt update
rosdep install -y --from-paths /path/to/AutowareArchitectureProposal/src/ --ignore-src --rosdistro foxy
```
6. Build and source the repository (the build will `stderr` see Discussion Points below)
```
colcon build
. install/setup.bash
```
7. Run the launch files and check the topics, parameters and node exists (two launch files total)
```
ros2 launch gnss_poser gnss_poser.launch.xml
ros2 launch gnss_poser ubloxfix2mgrs.launch.xml
```

## Discussion Points

### Logging
Non-ROS-node helper method originally had some ROS logging method calls. To convert this to ROS2 while maintaining the ROS-based logging I needed to pass the node's logger by reference to the methods. Not sure if this is the tried and vetted method of maintaining logging in non-ROS helper methods. Please advise.

### Ublox dependencies
The Ublox dependencies are stored in the `/sensing/driver/ublox` but in the AutowareArchitectureProposal repo the `.repos` file maps the ublox repository to a folder much higher. I noticed that the ublox repository is neither a submodule or pulled via the `.repos` is there a way we want to try and pull this resource?
